dotCMS/src/main/java/com/dotcms/util/AssetExporterUtil.java

```diff
@@ -0,0 +1,185 @@
+import com.dotmarketing.business.Role;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.ConfigUtils;
+import com.dotmarketing.util.DateUtil;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.StringUtils;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import com.liferay.util.FileUtil;
+import io.vavr.Lazy;
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import io.vavr.control.Try;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.io.IOUtils;
+
+import javax.ws.rs.core.MediaType;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileFilter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.StringWriter;
+import java.nio.file.Files;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Utility class for exporting assets.
+ *
+ * @author vic
+ */
+public class AssetExporterUtil {
+
+    enum EXPORT_VERSION {
+        LIVE, LIVE_WORKING;
+    }
+
+    static final String LIVE_WORKING_SQL =
+                    "select working_inode, live_inode from contentlet_version_info where working_inode > ? order by" +
+                            " working_inode limit 10000";
+    static final String[] ACCEPTED = new String[] {
+            "/0/",
+            "/1/",
+            "/2/",
+            "/3/",
+            "/4/",
+            "/5/",
+            "/6/",
+            "/7/",
+            "/a/",
+            "/b/",
+            "/c/",
+            "/d/",
+            "/e/",
+            "/f/",
+            "dotcms.sql.gz"
+    };
+    static final Lazy<File> PARENT = Lazy.of(() -> new File(ConfigUtils.getAbsoluteAssetsRootPath()));
+    static final FileFilter FILE_FILTER = pathname -> {
+        for (String dir : ACCEPTED) {
+            if ((pathname.getPath()+"/").contains(dir)) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    /**
+     * Utility private and empty constructor.
+     */
+    private AssetExporterUtil() {}
+
+    public static String exportToFile(final User user, final OutputStream outputStream) {
+        final String hostName = Try
+                .of(() -> APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), false).getHostname())
+                .getOrElse("dotcms");
+        final String fileName = StringUtils.sanitizeFileName(hostName)
+                + "_assets_"
+                + DateUtil.EXPORTING_DATE_FORMAT.format(new Date())
+                + ".tar.gz";
+        Logger.info(AssetExporterUtil.class, "User : " + user.getEmailAddress() + " downloading assets");
```

Comment:
I would do this as SecurityLog logger statement

Reply:
Done.